### ISSUE TITLE
Question about softmax score

### ISSUE THREAD

**Issue by quantaji** [concept-fusion/concept-fusion#3]
First thank you for this exciting paper! I have a huge question!

From my experience, with clip and open_clip, their similarity (img-img, img-text) from cosine similarity is very similar, that is also why the logits are multiplied with a scale factor (1/0.07) before passing to InfoNCE loss.

However, I saw this implementation, and there is not scale factor, this will result in a neasrly identical `softmax_scores`, and the weight factor for global feature is nearly $\frac{1}{R}$, where $R$ is the number of region proposal.

My question is two fold:
1. If the weight factor ($w_i$ in the original paper) is not very informative, is it still functional? 
2. I saw in the README.md, that you switch from mask2former to SAM. I checked that SAM gives a variable number of region proposal. This makes the $R$ varies for each frame. Will it cause instability?

Thanks!
